Indicators/PivotPointsHighLow.cs

```diff
@@ -0,0 +1,248 @@
+        public PivotPoint[] GetLowPivotPointsArray()
+        {
+            return _pivotPoints?.Where(p => p.PivotPointType == PivotPointType.Low).ToArray();
+        }
+
+        /// <summary>
+        /// Get all pivot points, in the order such that first element in collection is the nearest to the present date
+        /// </summary>
+        /// <returns>An array of low and high pivot points. Ordered by time in descending order.</returns>
+        /// <remarks>Returned array can be empty if no points have been registered yet/</remarks>
+        public PivotPoint[] GetAllPivotPointsArray()
+        {
+            // Get the copy of array
+            return _pivotPoints?.ToArray();
+        }
+
+        /// <summary>
+        /// Invokes NewPivotPointFormed event
+        /// </summary>
+        /// <param name="pivotPointsEventArgs"></param>
+        protected virtual void OnNewPivotPointFormed(PivotPointsEventArgs pivotPointsEventArgs)
+        {
+            NewPivotPointFormed?.Invoke(this, pivotPointsEventArgs);
+        }
+
+        // Will replace an array passed by reference with a new one, containing new pivot point,
+        // where the the order of the elements will be from nearest to farthest in time
+        private static void Push(PivotPoint point, ref PivotPoint[] array)
+        {
+            PivotPoint[] tempArray;
+            int size;
+            if (array == null)
+            {
+                size = 1;
+                tempArray = new PivotPoint[size];
+            }
+            else
+            {
+                size = array.Length + 1;
+                tempArray = new PivotPoint[size];
+                for (var n = 0; n < array.Length; n++)
+                {
+                    tempArray[n + 1] = array[n];
+                }
+            }
+
+            tempArray[0] = point;
+            array = tempArray;
+        }
+    }
+
+    /// <summary>
+    /// Represents the points identified by Pivot Point High/Low Indicator.
+    /// </summary>
+    public class PivotPoint : BaseData
+    {
+        /// <summary>
+        /// Represents pivot point type : High or Low
+        /// </summary>
+        public PivotPointType PivotPointType { get; set; }
+
+        /// <summary>
+        /// Peak value
+        /// </summary>
+        public sealed override decimal Value { get; set; }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="PivotPoint"/>
+        /// </summary>
+        public PivotPoint(PivotPointType type, decimal price, DateTime time)
+        {
+            PivotPointType = type;
+            Value = price;
+            Time = time;
+        }
+    }
+
+    /// <summary>
+    /// Pivot point direction
+    /// </summary>
+    public enum PivotPointType
+    {
+        /// <summary>
+        /// High pivot point
+        /// </summary>
+        High = 1,
+
+        /// <summary>
+        /// Low pivot point
+        /// </summary>
+        Low = -1
```

Comment:
Please add `Both`, and `None` here too so the result of the indicator can be parsed as an enum.